File: Oracle/metrics/registry.py
from .eth_staking import ETHStakingRateMetric
from .eth_price import ETHPriceMetric

class MetricRegistry:
    """
    Central registry for all metric handlers.
    """
    def __init__(self):
        self._metrics = {}
        # Register default metrics
        self.register(ETHStakingRateMetric())
        self.register(ETHPriceMetric())

    def register(self, metric):
        """Registers a new metric handler."""
        self._metrics[metric.name] = metric
        print(f"✅ Registered metric: {metric.name}")

    def get_metric(self, name):
        """Retrieves a metric handler by name."""
        return self._metrics.get(name)

    def list_metrics(self):
        """Returns a list of all registered metric names."""
        return list(self._metrics.keys())

# Global registry instance
registry = MetricRegistry()

File: Oracle/src/metrics/registry.ts
import { Metric } from "./base.js";
import { ETHStakingRateMetric } from "./eth_staking.js";
import { ETHPriceMetric } from "./eth_price.js";
import { BTCDominanceMetric } from "./btc_dominance.js";
import { ETHGasPriceMetric } from "./eth_gas_price.js";

class MetricRegistry {
    private metrics: Map<string, Metric> = new Map();

    constructor() {
        this.register(new ETHStakingRateMetric());
        this.register(new ETHPriceMetric());
        this.register(new BTCDominanceMetric());
        this.register(new ETHGasPriceMetric());
    }

    register(metric: Metric): void {
        this.metrics.set(metric.name, metric);
        console.log(`✅ Registered metric: ${metric.name}`);
    }

    getMetric(name: string): Metric | undefined {
        return this.metrics.get(name);
    }

    listMetrics(): string[] {
        return Array.from(this.metrics.keys());
    }
}

export const registry = new MetricRegistry();
